agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1315,28 +1315,30 @@ func (dg *dockerGoClient) APIVersion() (dockerclient.DockerVersion, error) {
 }
 
 // Stats returns a channel of *types.StatsJSON entries for the container.
-func (dg *dockerGoClient) Stats(ctx context.Context, id string, inactivityTimeout time.Duration) (<-chan *types.StatsJSON, error) {
+func (dg *dockerGoClient) Stats(ctx context.Context, id string, inactivityTimeout time.Duration) (<-chan *types.StatsJSON, <-chan error) {
 	subCtx, cancelRequest := context.WithCancel(ctx)
 
 	client, err := dg.sdkDockerClient()
 	if err != nil {
 		cancelRequest()
-		return nil, err
+		return nil, nil
 	}
 
-	// Create channel to hold the stats
-	statsChnl := make(chan *types.StatsJSON)
+	statsC := make(chan *types.StatsJSON)
+	errC := make(chan error)
 	var resp types.ContainerStats
 
 	if !dg.config.PollMetrics {
+		// Streaming metrics is the default behavior
+		seelog.Infof("DockerGoClient: Starting streaming metrics for container %s", id)
```

Comment:
can't this just be debug log?

Reply:
Corresponding log message for polling metrics below is Info, this shouldn't log very frequently, only once per container under normal circumstances so I'd prefer it to be info.